### PR TITLE
fix(rust,python): OOB in list set operations on empty frame

### PR DESCRIPTION
**Pull request by Object905**
Would fix https://github.com/pola-rs/polars/issues/12711

### CODE REVIEW

**Comment by ritchie46** [comment]
Thanks. I think we should just check the input series and if that is empty return it directly. That would ensure we never hit the code on empty cases.

**Comment by Object905** [comment]
Thanks for awesome library)

Yes, thats what though of initially, but this is a bit more involved for me. And this bug prevents me to upgrading to latests version.
Also this OOB happens even before `set_operation `is called and I honestly don't even understand logic outside of `set_operation`, as I'am not familiar with codebase :)

Generaly I think this special cases should be handled with regards to set operation logic.
Like for intersection - if any list empty - result is always empty, for union - if any list is empty - result is other list, etc.

**Comment by cmdlineluser** [comment]
I think this is the place you could check `.len()` 

https://github.com/pola-rs/polars/blob/ad1d2d9d5e7488b65b484d89a63c5dbd1eadd481/crates/polars-plan/src/dsl/function_expr/list.rs#L516-L521

(There are other examples in that file of len checks.)

**Comment by ritchie46** [comment]
Yeap, if s0 is empty we should return s0.

File: crates/polars-plan/src/dsl/function_expr/list.rs [hunk 1]
```diff
@@ -517,6 +517,29 @@ pub(super) fn unique(s: &Series, is_stable: bool) -> PolarsResult<Series> {
 pub(super) fn set_operation(s: &[Series], set_type: SetOperation) -> PolarsResult<Series> {
     let s0 = &s[0];
     let s1 = &s[1];
+
+    if s0.len() == 0 || s1.len() == 0 {
+        match set_type {
+            SetOperation::Intersection => {
+                if s0.len() == 0 {
+                    return Ok(s0.clone());
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/function_expr/list.rs:525]: nit: factor out `return` at match level.

```rust
return match {
..
}
```

**Review by ritchie46 — COMMENTED** [review]
Looks good. One small nitpick.

File: crates/polars-plan/src/dsl/function_expr/list.rs (hunk 1 above)
Comment (Object905) [reply crates/polars-plan/src/dsl/function_expr/list.rs]: done

**Review by ritchie46 — APPROVED** [review]
Thanks you @Object905

**Pull request merged**